pkg/internal/checkup/vmi.go

```diff
@@ -54,86 +54,70 @@ const (
 )
 
 func newVMIUnderTest(checkupConfig config.Config) *kvcorev1.VirtualMachineInstance {
-	vmiConfig := DPDKVMIConfig{
-		NamePrefix:                      VMIUnderTestNamePrefix,
-		OwnerName:                       checkupConfig.PodName,
-		OwnerUID:                        checkupConfig.PodUID,
-		Affinity:                        Affinity(checkupConfig.DPDKNodeLabelSelector, checkupConfig.PodUID),
-		ContainerDiskImage:              checkupConfig.VMContainerDiskImage,
-		NetworkAttachmentDefinitionName: checkupConfig.NetworkAttachmentDefinitionName,
-		NICEastMACAddress:               checkupConfig.DPDKEastMacAddress.String(),
-		NICEastPCIAddress:               config.VMIEastNICPCIAddress,
-		NICWestMACAddress:               checkupConfig.DPDKWestMacAddress.String(),
-		NICWestPCIAddress:               config.VMIWestNICPCIAddress,
-		Username:                        config.VMIUsername,
-		Password:                        config.VMIPassword,
-	}
+	vmiUnderTest := newBaseVMI(checkupConfig)
```

Comment:
It seems redundant to iterate over the options in two places (`vmi.New` and `applyOptionsToVMI`).
How about creating a function baseVMIOptions and then something like this?
```
baseOptions := baseVMIOptions(checkupConfig)
vmiunderTestCustomOptions := []vmi.Option{
		vmi.WithAffinity(Affinity(checkupConfig.DPDKNodeLabelSelector, checkupConfig.PodUID)),
		vmi.WithSRIOVInterface(eastNetworkName, checkupConfig.DPDKEastMacAddress.String(), config.VMIEastNICPCIAddress),
		vmi.WithSRIOVInterface(westNetworkName, checkupConfig.DPDKWestMacAddress.String(), config.VMIWestNICPCIAddress),
		vmi.WithContainerDisk(rootDiskName, checkupConfig.VMContainerDiskImage),
		vmi.WithCloudInitNoCloudVolume(cloudInitDiskName, CloudInit(config.VMIUsername, config.VMIPassword)
}

return vmi.New(RandomizeName(VMIUnderTestNamePrefix),baseOptions, vmiunderTestCustomOptions)
```

Reply:
Good idea.

Reply:
Done.
It really improved the code.
Thank you.